### DISCUSSION TITLE
Testing Patterns for referencing modules

### DISCUSSION THREAD

**Discussion by landau** [Urigo/graphql-modules#2129] [Q&A]
The team I am working with has been using `graphql-modules` project for almost a year (starting with V1 now on V2). We thank you and other committers for the efforts put in to the project. We have a concern regarding _modularization and testing_. In particular, when extending interfaces/types outside of a module's scope. We appreciate any insights you may have regarding our concern.

Please see this PR for more context and example code: https://github.com/landau/graphql-modules/pull/1

**Comment by landau** [comment]
👋 Just checking in here. Is this the ideal place to raise concerns? Thank you.

**Comment by landau** [comment]
Just checking in here again. Seeking some guidance.

**Comment by Brian-McBride** [comment]
I'm having issues with `testkit` even running resolvers. Like a todo app:

```typescript
    const app = testkit.testModule(todosGraphqlModule);
    expect(app.schema.getQueryType()).toBeDefined();
    expect(app.schema.getMutationType()).toBeDefined();

    const graphResult = await testkit.execute(app, {
      document: gql`
        query getTodo($id: ID!) {
          id
          title
        }
      `,
      variableValues: {
        id: 'someId',
      },
    });

    console.log(graphResult);  // output is:  { data {} }
```

Seems pretty simple. If I run the code in my app, the resolvers run as expected. But running this in Jest I always get ` { data {} } ` as the result, and the resolvers are never executed. 

But it isn't just `graphql-modules`; in general, unit/integration testing tooling has been lacking in every GraphQL lib I've used so far.  A super basic schema is simple - but as soon as you start extending types or use something like Apollo's Federation...

**Reply by ardatan** [reply]
If you share a reproduction on CodeSandbox or Stackblitz, we would love to help you. We need to see your schema together with your resolvers to reproduce and debug your issue. Thanks

**Reply by Brian-McBride** [reply]
@ardatan  I would love some advice here. 
I set up a "simple" example. The actual module is super simple, but I did set it up in an example production mono-repo setting. While the project is more complex in a general setup - the actual `jest` call is what is very simple.

The basic problem is that `testkit.execute` isn't calling the resolver function at all.

https://github.com/Brian-McBride/graphql-modules-testing-example

The [schema link](https://github.com/Brian-McBride/graphql-modules-testing-example/blob/cf5b89dc4eb555578d6240ab98b1805fe2001fc0/libs/server/modules/users/src/lib/schema/users.graphql.ts)

The [module link
](https://github.com/Brian-McBride/graphql-modules-testing-example/blob/cf5b89dc4eb555578d6240ab98b1805fe2001fc0/libs/server/modules/users/src/lib/users.module.ts)

The test is [here](https://github.com/Brian-McBride/graphql-modules-testing-example/blob/cf5b89dc4eb555578d6240ab98b1805fe2001fc0/libs/server/modules/users/src/lib/users.module.spec.ts):

**Reply by Brian-McBride** [reply]
@ardatan 
I added in another project in stackblitz, just the same thing copied and pasted over - however this structure is very basic. 

https://stackblitz.com/edit/node-mnkhc1?file=src/modules/users/users.module.spec.ts

Same issue. Just install the node modules if needed and `npm run test` and it will run that single test. 
I didn't bother putting in all the other code, so there is no actual graphql main app. Just the module and the test that returns an empty object when we should be seeing a value from the resolver.
